src/components/model/todo/TodoInput.tsx

```diff
@@ -0,0 +1,44 @@
+import clsx from "clsx";
+import type { ForwardRefExoticComponent, KeyboardEvent, RefAttributes } from "react";
+import { forwardRef } from "react";
+
+type TodoInputProps = {
+  categoryColor: string;
+  value: string;
+  onBlur: () => void;
+  onChange: () => void;
+  onEnterKeyPress: (e: KeyboardEvent<HTMLInputElement>) => void;
+  className?: string;
+};
+
+type TodoInput = ForwardRefExoticComponent<TodoInputProps & RefAttributes<HTMLInputElement>>;
+
+const checkedRadioBgTheme = (categoryColor: string) => {
+  return `radio-${categoryColor}`;
+};
+const caretTheme = (categoryColor: string) => {
+  return `caret-${categoryColor}`;
+};
+
+export const TodoInput: TodoInput = forwardRef((props, ref) => {
```

Comment:
```suggestion
export const TodoInput = forwardRef<HTMLInputElement, TodoInputProps>((props, ref) => {
```

Genericsの型引数の順番が実際のcallbackの引数の順番と逆になっていてわかりづらいのですが、
このようにGenericsで指定してやれば型注釈は必要なくなると思います🤔

ライブラリなどから提供されている機能を用いる際に型をつけたい場合、たいていはGenericsとして
指定できるメカニズムが用意されているという認識でよいかな〜と思います！

(逆に修正前のやつで、実装できてしまうのがすごいです👏)

Reply:
勉強になります、ありがとうございます！ 🙏 
修正しました！